src/index.js

```diff
@@ -72,6 +84,26 @@ var Index = new Lang.Class({
             doc.add_numeric_value(SEQUENCE_NUMBER_VALUE_NO, metadata['sequenceNumber']);
         }
 
+        const indexingDateString = _indexDateForMetadata(metadata);
+        if (indexingDateString) {
+            /* Parse the date using the native JS Date library, since
+             * GLib.Date.set_parse isn't able to handle JS-style dates. */
+            try {
+                const parsedIndexingDate = new Date(indexingDateString)
+
+                const timestamp = Number([parsedIndexingDate.getUTCFullYear(),
```

Comment:
Is this what we do in xapianlib? It seems like we should use Unix timestamps here, but it's more important to be consistent.

Reply:
...and if this is what we do in xapianlib, it seems like we are doing it very wrong:
```
gjs> let date = new Date('January 1, 2018 01:01:01 UTC');
gjs> Number([date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDay(), date.getUTCHours(), date.getUTCMinutes(), date.getUTCSeconds()].join(''))
201801111
gjs> date = new Date('December 31, 2018 12:12:12 UTC');
Mon Dec 31 2018 04:12:12 GMT-0800 (PST)
gjs> Number([date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDay(), date.getUTCHours(), date.getUTCMinutes(), date.getUTCSeconds()].join(''))
2018111121212
```

Reply:
This is kind of what happens in xapianlib, though there we encode each element with a fixed length (eg 01). I'll see if the same can be done here.

Reply:
Any idea why we do this in xapianlib?

Heads up, you should probably also check the month number between this code and xapianlib. JS's months start with January = 0.

Reply:
If I remember correctly, it was because the inbuilt date-to-number serialization method in Python crashed if the date was prior to 1 January 1930, so we had to serialize by hand.

In any case JS months starting with 0 is a good point. I'll cross-check how it works in Python.

Reply:
Hey @smspillaz did you get to compare this?

I did a quick test, here what we actually pass to doc.add_value/add_numeric_value:

For `2018-01-13T12:46:09.000Z`
here we get: `20180113124609`
xapianlib gets: `2.01801131246e+13`

Technically not the same, but I don't think it should be a problem. @ptomato what you think?

Reply:
Are we actually running into the precision of the `double` type or is that to do with the output precision that you're printing it with? I thought doubles had more precision than that.

Reply:
you are right, its '20180113124609.000000'

`20180113124609.000000`
`20180113124609`